### ISSUE TITLE
Invalid UTF-8 characters in hover title

### ISSUE THREAD

**Issue by HayoBaan** [doxygen/doxygen#6708]
While the warning from https://github.com/doxygen/doxygen/issues/6370 is indeed gone in 1.8.15, the text shown in the hover in the dependency graph still has the UTF-8 characters showing incorrectly e.g. `“true”` gets replaced with `âtrueâ` (c3 a2 c2 80 c2 9c 74 72  75 65 c3 a2 c2 80 c2 9d in hex).

For an example, see:
[doxytest_utf8.zip](https://github.com/doxygen/doxygen/files/2715084/doxytest_utf8.zip)

**Comment by albert-github** [comment]
Problem looks like to be in the conversion to png when using svg it looks OK (as far as I can see). 
png situation has to be investigated further.

**Comment by albert-github** [comment]
I had a closer look and with the dot 2.38.0 version (on Windows, dot -V) I get the problems as indicated, with the version 2.40.1 (on Cygwin) I saw the right tooltips, so I think the problem is the Graphviz dot version.

I found a version 2.40.1 for Cygwin, but I haven't been able to locate a native 2.40.1 (or newer) version for Windows.

**Comment by HayoBaan** [comment]
Interesting, so you're saying, the tooltip is generated by graphviz not doxygen itself?

However, as I'm already on 2.40.1 the version alone doesn't explain the issue. I'm on a Mac though, not on windows.

Note: the correct encoding for `“` would be e2 80 9c which gets translated into c3 a2 c2 80 c2 9c which is the double encoded form of e2 80 9c… So somewhere strings are getting doubly encoded…

**Comment by albert-github** [comment]
I re-evaluated and you are right (I have no Mac so I cannot test it there, but looks like the platform is not relevant).
In the 1.8.14 it works with dot 2.38.0 and 2.40.1.
Problem was that the default version for me on Cygwin is 1.8.14 and on Windows 1.8.15.
I now ran both (Windows and Cygwin so 2.38.0 and 2.40.1) with 1.8.14 and it is OK and with 1.8.15 there is a problem.
Interesting is that the basic information (dot and map file) looks OK but the `area` information in the html file isn't (1.8.16 is in my case still identical to 1.8.15):
```
>diff -r -b -w html_windows_1_8_14/file1_8c.html html_windows/file1_8c.html

32c32
< <!-- Generated by Doxygen 1.8.14 -->
---
> <!-- Generated by Doxygen 1.8.16 -->
78,79c78,80
< <area shape="rect" id="node2" href="file1_8h.html" title=""Test" header. " alt="" coords="5,80,61,107"/>
< <area shape="rect" id="node3" href="file2_8h.html" title=""Include" file2. " alt="" coords="85,80,141,107"/>
---
> <area shape="rect"  title="â??Testâ?? source." alt="" coords="45,5,101,32"/>
> <area shape="rect"  href="file1_8h.html" title="â??Testâ?? header." alt="" coords="5,80,61,107"/>
> <area shape="rect"  href="file2_8h.html" title="â??Includeâ?? file2." alt="" coords="85,80,141,107"/> 
```

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #6719

**Comment by albert-github** [comment]
Code has been integrated in master on github.

**Comment by HayoBaan** [comment]
Awesome, just built the latest code and can confirm the issue solved 😃

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by HayoBaan** [comment]
While the issue was fixed earlier (e.g. 536355d1), the problem with 1.8.16 is that now we have **no** hover text at all, **nowhere**!

I looked at the changes and identified 6c76841e the commit that caused the problem (87597541 still shows the hovers)
